homeassistant/components/edl21/config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Config flow for EDL21 integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_SERIAL_PORT, DEFAULT_TITLE, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_SERIAL_PORT): str,
+    }
+)
+
+
+class EDL21ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """EDL21 config flow."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+
+        self._async_abort_entries_match(
+            {CONF_SERIAL_PORT: import_config[CONF_SERIAL_PORT]}
+        )
+        if import_config[CONF_NAME]:
+            self._async_abort_entries_match({CONF_NAME: import_config[CONF_NAME]})
+        return self.async_create_entry(
+            title=import_config[CONF_NAME] or DEFAULT_TITLE,
+            data=import_config,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Handle the user setup step."""
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {CONF_SERIAL_PORT: user_input[CONF_SERIAL_PORT]}
+            )
+
+            return self.async_create_entry(
+                title=DEFAULT_TITLE,
+                data=user_input,
+            )
```

Comment:
Is there some kind of "try_connect" that could be implemented to ensure the port is correct?
Maybe some kind of:
```python
    protocol = SmlProtocol(serial_port):
    try:
        connect()
    except SomeException:
        errors["base"] = "cannot_connect"
    else:
        return self.async_create_entry
```

Reply:
It looks like the exceptions during the connection process are caught internally and are not "rethrown" (https://github.com/mtdcr/pysml/blob/a555fb4f8de123d383e6aea733d7f0b39b57c3fc/sml/asyncio.py#L114). I think this is not possible without additional changes to the pysml dependency. Maybe @mtdcr is open to discuss this. :)